### DISCUSSION TITLE
-proxy-internal -pi flag not work

### DISCUSSION THREAD

**Discussion by djallalzoldik** [projectdiscovery/nuclei#3741] [Q&A]
when i using nuclei like this 
nuclei -l list.txt -t screen.yaml -headless -pi "http://127.0.0.1:8080"

the traffic should go to my burp suite, but is not i don't know why 
nothing go to my brup suite , 

can you show me how this flag work?

when i start the command this what will happen 
![local](https://github.com/projectdiscovery/nuclei/assets/45321671/bd385d2c-7fd2-41b0-8b2f-eeec9110cb29)

**Comment by MetzinAround** [comment]
hi @djallalzoldik . I'm gonna tag @ehsandeep 

Is this expected behavior? Should localhost go to burpsuite when it's scanned?

**Reply by djallalzoldik** [reply]
I hope to get answer from him

**Reply by olearycrew** [reply]
@djallalzoldik I wonder if you wanted to use the `-p` flag for "proxy" vs. the `-pi` flag for "proxy internal"

Proxy Internal is a bool for true/false whereas `-p` takes a string (or list) of http/socks5 proxies to use.

**Reply by djallalzoldik** [reply]
@olearycrew ,I don't understand what is the role of -pi
I thought he was sending all requests through the internal proxy. To clarify the idea, I currently have an idea that it plays a role like  Foxy Proxy exstension? am i wrong?

**Comment by ehsandeep** [comment]
@djallalzoldik if you are looking to use proxy, use `-p` option, `-pi` is optional flag to enable proxying traffic from nuclei unrelated to scan like reporting integration to GitHub/Jira etc.

As per your description, this should be valid example -

```
nuclei -l list.txt -t screen.yaml -headless -p "http://127.0.0.1:8080"
```
